applications/luci-app-xfrpc/htdocs/luci-static/resources/view/xfrpc.js

```diff
@@ -0,0 +1,176 @@
+				    '<svg class="gh-button__icon gh-button__icon--star" viewbox="0 0 896 1024">' +
+				    '<path d="M896 384l-313.5-40.781L448 64 313.469 343.219 0 384l230.469 208.875L171 895.938l277-148.812 277.062 148.812L665.5 592.875 896 384z" />' +
+				    '</svg>' +
+				    '<span class="gh-button__stat__text">4,500</span>' +
+				  	'</span>' +
+					'</a>';
+	
+	renderHTML += starHTML;
+	
+	return renderHTML;
+}
+
+document.querySelector('head').appendChild(E('link', {
+		    'rel': 'stylesheet',
+		    'type': 'text/css',
+		    'href': L.resource('view/xfrpc/gh-button.css')
+}));
+
+return view.extend({
+	render: function() {
+		var m, s, o;
+
+		m = new form.Map('xfrpc', _('xfrpc'));
+		m.description = _("xfrpc is a c language frp client for frps. It has more advantage in OpenWrt box than frpc");
+		
+		s = m.section(form.NamedSection, '_github');
+		s.anonymous = true;
+		s.render = function (section_id) {
+			L.Poll.add(function () {
+				return L.resolveDefault().then(function() {
+					var view = document.getElementById("github_info");
+					view.innerHTML = renderGithubInfo();
+				});
+			});
+
+			return E('div', { class: 'cbi-map' },
+				E('fieldset', { class: 'cbi-section'}, [
+					E('p', { id: 'github_info' },
+						_('Collecting x-frpc github info ...'))
+				])
+			);
+
+		}
+
+		s = m.section(form.NamedSection, '_status');
+		s.anonymous = true;
+		s.render = function (section_id) {
+			L.Poll.add(function () {
+				return L.resolveDefault(getServiceStatus()).then(function(res) {
+					var view = document.getElementById("service_status");
+					view.innerHTML = renderStatus(res);
+				});
+			});
+
+			return E('div', { class: 'cbi-map' },
+				E('fieldset', { class: 'cbi-section'}, [
+					E('p', { id: 'service_status' },
+						_('Collecting data ...'))
+				])
+			);
+		}
+
+		s = m.section(form.NamedSection, 'common', 'xfrpc');
+		s.dynamic = true;
+
+		s.tab('common', _('Common Settings'));
+		s.tab('init', _('Startup Settings'));
+
+		o = s.taboption('common', form.Value, 'server_addr', _('Server address'), _('Server address specifies the address of the server to connect to.<br />By default, this value is "0.0.0.0".'));
+		o = s.taboption('common', form.Value, 'server_port', _('Server port'), _('Server port specifies the port to connect to the server on.<br />By default, this value is 7000.'));
+		o = s.taboption('common', form.Value, 'token', _('Token'), _('Token specifies the authorization token used to create keys to be sent to the server. The server must have a matching token for authorization to succeed. <br />By default, this value is "".'));
+
+		o = s.taboption('init', form.Flag, 'disabled', _('Disabled xfrpc service'));
+		o = s.taboption('init', form.ListValue, 'loglevel', _('Log level'), _('LogLevel specifies the minimum log level. Valid values are "debug", "info", "notice", "warning", "err", "crit", "alert" and "emerg".<br />By default, this value is "info".'));	
+		o.value('7');
```

Comment:
I think here it would be better to have not only the loglevel as a number as a dropdown, but a string for example `Debug`, `Warning` and so on. Check this repository to see how other modules do this.

Reply:
@feckert  could u give me an example?